### PR TITLE
Wallet e2e on Circle Ci

### LINKED ISSUE #146: Revive Wallet app E2E tests
### Expected Behavior

E2E tests run on a loop and post result somewhere noticeable

### Current Behavior

E2E tests not running on a regular schedule

### PR DESCRIPTION
**Pull request by martinvol**
### Description

e2e runs on Circle CI

### Tested

itself.

### Other changes

- 

### Related issues

- Fixes #146

### Backwards compatibility

-

### CODE REVIEW

**Comment by cmcewen** [comment]
I hope there is some way we can cache the deps like android tools and node here

File: .circleci/config.yml [hunk 1]
```diff
@@ -104,6 +117,125 @@ jobs:
...
+            # TODO, Make cache work and figure out why this doesn't work
+            # if [ ! -d node_modules ]; then
+            #     echo "No cache found, install yarn. Hold tight."
+            #     yarn
+            # else
+            #   echo "Cache found, just run post-script."
+            #   yarn postinstall
+            # fi
+            yarn
+            yarn build
```
Comment (martinvol) [inline .circleci/config.yml:198]: any idea why this fails with 
```
tsc
src/index.ts(1,43): error TS2307: Cannot find module '@celo/contractkit'.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lerna ERR! yarn run build stderr:
warning package.json: No license field
error Command failed with exit code 2.

lerna ERR! yarn run build exited 2 in '@celo/dappkit'
```
if comment the line above and uncomment the if block @ashishb?

Comment (ashishb) [reply .circleci/config.yml:198]: No idea. Let's get the working code in and fix caching later.

File: packages/mobile/scripts/unlock.sh [hunk 2]
```diff
@@ -18,7 +18,13 @@ done
 adb wait-for-device shell \
   'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done;'
 
+
+echo "locksettings set-pin 123456" | adb shell
```
Comment (ashishb) [inline packages/mobile/scripts/unlock.sh:22]: `adb shell locksettings set-pin 123456` might work as well

File: packages/mobile/scripts/unlock.sh [hunk 3]
```diff
@@ -18,7 +18,13 @@ done
...
+
+sleep 1
 echo "Device is done booting"
+sleep 10
```
Comment (ashishb) [inline packages/mobile/scripts/unlock.sh:26]: Why sleep?

**Review by ashishb — APPROVED** [review]
Better to move testing script into `e2e` directory. Why keep them in  a separate scripts directory?

File: packages/mobile/scripts/unlock.sh (hunk 3 above)
Comment (martinvol) [reply packages/mobile/scripts/unlock.sh:26]: Just to make sure the device is responding to keyboard events before trying to unlocking

**Comment by martinvol** [comment]
Thanks to @kamyar for the foundational work on this!

**Comment by kamyar** [comment]
Wow! Amazing work! 🚀 
Happy to hear it helped 💃 and kudos for taking it past the finish line, [CI](https://circleci.com/gh/celo-org/celo-monorepo/40988?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) looks amazing! 😄 

p.s. Connor was instrumental in getting it to build with detox in the first place. 😅 

\cc @martinvol @ashishb @cmcewen

**Pull request merged**